redfish-core/lib/assembly.hpp

```diff
@@ -711,6 +712,224 @@ inline void getChassis(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                            << "] to fill in the URI property";
+                        messages::internalError(aResp->res);
+                        return;
+                    }
+
+                    if (subtrees.empty())
+                    {
+                        BMCWEB_LOG_ERROR
+                            << "No subtree in the ["
+                            << assemblyParentObjPath.str
+                            << "] to check assembled objects implementation "
+                            << "to fill the assembled object [ "
+                            << assembledObjPath.str
+                            << "] id in the URI property";
+                        messages::internalError(aResp->res);
+                        return;
+                    }
+
+                    std::vector<std::string> implAssemblyAssocs;
+                    for (const auto& subtree : subtrees)
+                    {
+                        auto it = std::find(assemblyAssoc.begin(),
+                                            assemblyAssoc.end(), subtree.first);
+                        if (it != assemblyAssoc.end())
+                        {
+                            implAssemblyAssocs.emplace_back(*it);
+                        }
+                    }
+
+                    if (implAssemblyAssocs.empty())
+                    {
+                        BMCWEB_LOG_ERROR
+                            << "The assembled objects of the ["
+                            << assemblyParentObjPath.str
+                            << "] are not implemented so "
+                            << "unable to fill the assembled object [ "
+                            << assembledObjPath.str
+                            << "] id in the URI property";
+                        messages::internalError(aResp->res);
+                        return;
+                    }
+
+                    // sort the implemented assemply object as per bmcweb design
+                    // to match with Assembly GET and PATCH handler.
+                    std::sort(implAssemblyAssocs.begin(),
+                              implAssemblyAssocs.end());
+
+                    auto assembledObjectIt = std::find(
+                        implAssemblyAssocs.begin(), implAssemblyAssocs.end(),
+                        assembledObjPath.str);
+
+                    if (assembledObjectIt == implAssemblyAssocs.end())
+                    {
+                        BMCWEB_LOG_ERROR
+                            << "The assembled object [" << assembledObjPath.str
+                            << "] in the object [" << assemblyParentObjPath.str
+                            << "] is not implemented so unable to fill "
+                            << "assembled object id in the URI property";
+                        messages::internalError(aResp->res);
+                        return;
+                    }
+
+                    auto assembledObjectId = std::distance(
```

Comment:
#include <iterator> ?

Reply:
Ack, I thought we no need since bmcweb is a kind of header-specific implementation, not a CPP file because added and removed no effects but, I will add to avoid dependence on the other bmcweb header file.